idaes/models/costing/SSLW.py

```diff
@@ -1016,13 +1016,21 @@ def cost_compressor(
             integer: whether the number of units should be constrained to be
                 an integer or not (default = True).
         """
-        # Confirm that unit is a turbine
+        # Confirm that unit is a compressor
         if not blk.unit_model.config.compressor:
             raise TypeError(
                 "cost_compressor method is only appropriate for "
                 "pressure changers with the compressor argument "
                 "equal to True."
             )
+        # compressor = True, and using non-isentropic assumption
+        # (costing not needed)
+        elif hasattr(blk.unit_model.config, "thermodynamic_assumption"):
```

Comment:
1. Do we need the `hasattr` check here? What models do we expect not to have this attribute?
2. In the following line, I would check against the `Enum` object itself, not the name attribute of the `Enum`. I.e. ` if blk.unit_model.config.thermodynamic_assumption.name != ThermodynamicAssumption.isentropic`